Lib/blackrenderer/render.py

```diff
@@ -111,6 +114,13 @@ def calcGlyphLineBounds(glyphLine, font):
             bounds = glyphBounds
         else:
             bounds = unionRect(bounds, glyphBounds)
+    if useFontMetrics:
+        ttf = font.ttFont
```

Comment:
The correct glyph metrics (as provided by HarfBuzz) are right there in `glyphLine`. You need to sum `glyph. xAdvance` to get the final position. No need to use `ttFont`.